genie-agent/src/main/java/com/netflix/genie/agent/cli/JobRequestArgumentsImpl.java

```diff
@@ -156,4 +156,11 @@
         converter = ArgumentConverters.JSONConverter.class
     )
     private JsonNode jobMetadata = GenieObjectMapper.getMapper().createObjectNode();
+
+    @Parameter(
+        names = {"--jobSpecificationFile"},
+        description = "The local file where to load a job specification from, rather than resolving via server",
```

Comment:
How do you envision using this in production vs. the `getSpecification` one?

Reply:
When we submit jobs via API and start the agent inside a container we could use this.
It would ensure the spec is created by the server before we even attempt to fork a container, and once the agent comes up we avoid having to resolve again.